ParabolInc/parabol
# Make Slack notifications pretty

PR to resolve issue #4911. 

It also resolves an undocumented bug where timer adjustments send new notifications rather than update the previous one. I've commented inline about this. 

A few additional notes:

- The issue mentioned using Slack's default green `#007a5a` but with Slack's blocks, you choose whether it's a primary button, regular or danger button rather than specify the colour. On a dark mode theme, we get the darker button and in a light mode, it's a lighter button
- I haven't come across a good, free tool for creating short links so I've included the regular link without utm params which is still shortish
- I've added a dummy interactivity request URL to the [Slack app](https://api.slack.com/apps/A01EGHDU7PZ/interactive-messages?). Otherwise, a warning sign shows after clicking the button. See Slack issue [here](https://github.com/CircleCI-Public/slack-orb/issues/188#issuecomment-708988009)
- I noticed that the comment count is currently `undefined` when ending poker meetings. I've created an issue to fix this [here](https://github.com/ParabolInc/parabol/issues/4983). When it's fixed, we can add comment count to the poker notifications

### To test

- Go to the `.env` file and change the `SLACK_CLIENT_ID` & `SLACK_CLIENT_SECRET` to the client ID and secret found in the `Parabol-staging` bot [here](https://api.slack.com/apps/A01EGHDU7PZ)
- Kill and restart the server with `yarn dev`
- You may need to install the Parabol-staging Slackbot in Slack
- Integrate Slack in Parabol (remove it first if it's already integrated),
- See the new notifications styles when you create, set a timer & end a meeting
- If the notifications are sent to `@Parabol-staging` the previous notification should be updated when you adjust the timer rather than sending a new notification
